### PR TITLE
boundary tag support increment/decrement from start to end

### PR DESCRIPTION
**Pull request by jayeeliu**
boundary tag support increment/decrement from start to end

### CODE REVIEW

**Comment by jayeeliu** [comment]
@bxcodec Could you help me to review this pull request ?

**Comment by bxcodec** [comment]
Btw, What's the use-case for this PR? @jayeeliu

**Comment by jayeeliu** [comment]
@bxcodec 
To test assignment rules (such as assigning students to teachers according to rules), we need to generate non-repetitive and ordered IDs.

For example, in polling mode, 20 is allocated per round, i need to test  assignment process is as follows:
1-20 -> A
21-40 -> B
41-60 -> C
61-80 -> A
...

File: faker.go [hunk 1]
```diff
@@ -325,6 +331,11 @@ func SetRandomNumberBoundaries(start, end int) error {
 // FakeData is the main function. Will generate a fake data based on your struct.  You can use this for automation testing, or anything that need automated data.
 // You don't need to Create your own data for your testing.
 func FakeData(a interface{}) error {
+	// if parameter != pre-parameter reset stateful data
+	if a != currentFaker {
+		currentFaker = a
+		boundaryCurrentNumber = 0
+	}
```
Comment (bxcodec) [inline faker.go:338]: Isn't this going to be not safe for multi-thread processing?

Comment (jayeeliu) [reply faker.go:340]: fixed

Comment (jayeeliu) [reply faker.go:338]: Incr/Decr does not support multi-thread and multi-struct thread-safe operation, is a stateful processing.
Example struct A and B
fakeData(A) -> 1
fakeData(A) -> 2
fakeData(B) -> 1
fakeData(A) -> 1, excepted 3
fakeData(B) -> 1, excepted 2
I need to record all procedure data or lock n-m fakeData processing, i think that would be too complicated and ineffective.

Single struct and multi-thread operation locked by func stepIntegerWithBoundary (in facker.go line 1150-1152)
func stepIntegerWithBoundary(boundary numberBoundary) (int, error) {
	mu.Lock()
	defer mu.Unlock()

**Comment by bxcodec** [comment]
Hi @jayeeliu, we're working on v4. The biggest change is about `options`. I think your proposal here is very well suited to the new approach. Can you try to upgrade and follow?

**Pull request closed without merging**